wikimedia/wikipedia-ios
# Add Notifications Center status bar text

**Phabricator:** https://phabricator.wikimedia.org/T288874 (partial)

### Notes

This is a follow-up PR to https://github.com/wikimedia/wikipedia-ios/pull/4090 that updates the actual status bar text. As requested, I’ve included the necessary test data to expose the different permutations of the status bar to hopefully speed up testing.

Note the `TODO` included in `RemoteNotificationsController.swift`. This value (`totalLocalProjectsCount`) should eventually be populated/updated with the correct local project/inbox count. From related data work in the diary study prototype, I’m not sure where this value would actually end up living/be populated, but just added it here hardcoded for testing purposes. It’s free to eventually move to wherever it needs to be once the data side of things settles.

### Test Steps

In `RemoteNotificationsController.swift`, edit both `totalLocalProjectsCount` and the return value of `filterState` to display different attributed text permutations in the status bar. Note that this will also flow down to update the engaged/disengaged state of the filter and project bar buttons as needed. Confirm that the status bar is always populated with coherent strings. Here’s test data to return in `public lazy var filterState: RemoteNotificationsFilterState = { ... }`:

```swift
// No filtering
return RemoteNotificationsFilterState(readStatus: .all, types: [], projects: []) 

// Read status filtering only
return RemoteNotificationsFilterState(readStatus: .read, types: [], projects: [])
return RemoteNotificationsFilterState(readStatus: .unread, types: [], projects: [])

// Read status and single type filtering
return RemoteNotificationsFilterState(readStatus: .all, types: [.userTalkPageMessage], projects: []) 
return RemoteNotificationsFilterState(readStatus: .read, types: [.userTalkPageMessage], projects: [])
return RemoteNotificationsFilterState(readStatus: .unread, types: [.userTalkPageMessage], projects: [])

// Read status and project filtering
return RemoteNotificationsFilterState(readStatus: .all, types: [], projects: [.commons])
return RemoteNotificationsFilterState(readStatus: .read, types: [], projects: [.commons])
return RemoteNotificationsFilterState(readStatus: .unread, types: [], projects: [.commons])

// Read status and multiple type filtering
return RemoteNotificationsFilterState(readStatus: .all, types: [.userTalkPageMessage, .editMilestone, .mentionInTalkPage], projects: [])
return RemoteNotificationsFilterState(readStatus: .read, types: [.userTalkPageMessage, .editMilestone, .mentionInTalkPage], projects: [])
return RemoteNotificationsFilterState(readStatus: .unread, types: [.userTalkPageMessage, .editMilestone, .mentionInTalkPage], projects: [])

// Read status and multiple project filtering
return RemoteNotificationsFilterState(readStatus: .all, types: [], projects: [.commons, .wikidata])
return RemoteNotificationsFilterState(readStatus: .read, types: [], projects: [.commons, .wikidata])
return RemoteNotificationsFilterState(readStatus: .unread, types: [], projects: [.commons, .wikidata])

// Read status, multiple type, and multiple project filtering
return RemoteNotificationsFilterState(readStatus: .all, types: [.userTalkPageMessage, .editMilestone, .mentionInTalkPage], projects: [.commons, .wikidata])
return RemoteNotificationsFilterState(readStatus: .read, types: [.userTalkPageMessage, .editMilestone, .mentionInTalkPage], projects: [.commons, .wikidata])
return RemoteNotificationsFilterState(readStatus: .unread, types: [.editReverted, .userTalkPageMessage, .editMilestone, .mentionInTalkPage], projects: [.commons, .wikidata])
```
